$FreeBSD$

. Set default FreeBSD values for MAVEN_HOME and JAVA_HOME
. Use 'exec' to launch the JVM

--- bin/maven.orig	Tue Dec  7 12:13:46 2004
+++ bin/maven	Mon Jun 19 17:31:50 2006
@@ -19,6 +19,8 @@
 #   reserved.
 
 FOREHEAD_VERSION=1.0-beta-5
+: ${MAVEN_HOME="%%DATADIR%%"}
+: ${JAVA_HOME="`JAVAVM_DRYRUN=yes %%LOCALBASE%%/bin/java | %%GREP%% '^JAVA_HOME' | %%CUT%% -d= -f2`"}
 
 if [ -z "$MAVEN_OPTS" ] ; then
   MAVEN_OPTS="-Xmx256m"
@@ -150,7 +153,7 @@
   MAVEN_OPTS="$MAVEN_OPTS -Dmaven.home.local=${MAVEN_HOME_LOCAL}" 
 fi
   
-"$JAVACMD" \
+exec "$JAVACMD" \
   $MAVEN_OPTS \
   -Djavax.xml.parsers.SAXParserFactory=org.apache.xerces.jaxp.SAXParserFactoryImpl \
   -Djavax.xml.parsers.DocumentBuilderFactory=org.apache.xerces.jaxp.DocumentBuilderFactoryImpl \
